server/app.mjs

```diff
@@ -532,3 +454,85 @@ function gptscriptConfigPath() {
 
   return path.join(configDir, 'gptscript', 'config.json');
 }
+
+function pickToolName(toolName, existing) {
+  if (!toolName) {
+    toolName = 'external';
+  }
+
+  let testName = toolNormalizer(toolName);
+  while (existing.has(testName)) {
+    testName += '0';
+  }
+  existing.add(testName);
+  return testName;
+}
+
+function toolNormalizer(tool) {
+  const invalidChars = /[^a-zA-Z0-9_]+/g;
+  const validToolName = /^[a-zA-Z0-9]{1,64}$/;
+
+  let parts = tool.split('/');
+  tool = parts[parts.length - 1];
+  if (tool.endsWith('.gpt')) {
+    tool = tool.slice(0, -4);
+  }
+  tool = tool.replace(/^sys\./, '');
+
+  if (validToolName.test(tool)) {
+    return tool;
+  }
+
+  if (tool.length > 55) {
+    tool = tool.slice(0, 55);
+  }
+
+  tool = tool.replace(invalidChars, '_');
+
+  let result = [];
+  let appended = false;
+  for (let part of tool.split('_')) {
+    let lower = part.toLowerCase();
+    if (appended && lower.length > 0) {
+      lower = lower.charAt(0).toUpperCase() + lower.slice(1);
+    }
+    if (lower) {
+      result.push(lower);
+      appended = true;
+    }
+  }
+
+  let final = result.join('');
+  return final || 'tool';
+}
+
+function toChatStateTools(toolSet) {
+  const toolNames = new Set();
+
+  return Object.entries(toolSet)
+    .filter(([key]) => !key.startsWith('inline:'))
+    .map(([key, tool]) => {
+      let toolName = tool.name || key;
+      toolName = pickToolName(toolName, toolNames);
+
+      let args = tool.arguments;
+      if (!args) {
+        args = {}; // Default to an empty object if no arguments are provided
+      }
+
+      if (!tool.instructions) {
+        console.debug(`Skipping zero instruction tool ${toolName} (${key})`);
+        return null;
+      }
+
+      return {
+        function: {
+          toolID: key,
+          name: toolName,
+          description: tool.description,
+          parameters: args,
+        },
+      };
+    })
+    .filter((tool) => tool !== null);
```

Comment:
Nit: you can just add an additional filter 
```
.filter(([key, tool]) => return tool.instructions))
```

Reply:
good call! updated.